### ISSUE TITLE
[Feature request] Option to auto add single production items

### ISSUE THREAD

**Issue by LoneWolfProductions** [ShadowTheAge/yafc#148]
Let me try to describe it as good as possible.

When making a production sheet with big mods (ie : Pyanodons) you end up clicking alot of products that have only 1 assembly option.
Could we have a button that yafc would automaticly add this to the sheet as their clearly is no choice to be made?
Maybe also add the option to mark a product to not auto add if so desired (sheet bound)

I found in many cases I'm more clicking just to add it without it having any effect because of the product having only 1 production option.

This would be only affecting the current sheet to avoid your other sheets to be affected of course.

**Comment by ShadowTheAge** [comment]
But there is always a choice of not adding a production

**Comment by LoneWolfProductions** [comment]
True,

But since it would be optional, and you could mark it to be exclude before pressing the button it's fully controllable i think.

It would only add the ones currently in the "sumary ingredients" list.
Any new ones added as a result would not get processed yet until you press the button again after you visual conlucded you want them or mark them to be exclude.

(I hope that made some sence :p, i see it before me but explaining it over text.....)

**Comment by ShadowTheAge** [comment]
I will consider adding something like ctrl+click as a shortcut to "add production/consumption recipe that you see fits the best" but there are a bunch of corner cases

**Comment by LoneWolfProductions** [comment]
I was more think something like this

I pressed latex to be added. as you can see there is only one option. 
Instead of opening this and forcing me to click that single recipe, just add it.
after a while it becomes a bit of a drag to click these single recipes

I think we can safely assume i wanted to add it since i clicked it.
(and maybe an option in  the menu to turn this on/off if its the only one regarding the milestone settings)

![image](https://user-images.githubusercontent.com/30441722/151564848-d3de4c57-84ba-4802-b3b3-7cdf904ba07c.png)

**Comment by ShadowTheAge** [comment]
But in case in some future table you add something that requires latex that you can import and boom - you have a bunch of lines added that you need to now find and delete. That is a lot more frustrating than adding.

Also when adding recipes there is one more choice - which subtable do you want to add the recipe?

Because of these two cases, I will not add "fully automatic" recipe addition. There must be at least some user initiated action. I am okay with reducting the amount of clicks from 2 to 1 (instead of click on object - click on recipe, just ctrl+click on object)

**Comment by LoneWolfProductions** [comment]
ok it seems we are both on a different page.
i revised the idea im my last post from the original a bit

my final thoughts

the example i gave you has in the entire game only one option (and more like this are present)
sodium alinate (only one option to make no matter what) -> latex slab (same, one option) -> Latex (same, one option)

i do not understand what you mean with "subtable" unless you meant "normal add it to the sheet" or "add it to a nested table"
if thats the case then it depends where i clicked it. in the summary of the sheet or in the summary of the nested table should indicate where it should go then.

nothing would change just the second click would be gone because its redundant; i have only one choice anyway at that point in time.

if in a later stage something changed with the recipe (mod update) i would need to search them anyway to remove them if i want to adapt to the updated mod

if i need to inspect the tooltip to see if there's "only one option or not" to deceide what kind of click i want to give 
i can better just click it and the select the single recipe as this will not help in speed won but lost.

and i dont always want the best by yafc option. as this most of the time conflicts with my builds 
(many others stated this also, but thats another point)

Thank you for your time.

**Comment by ShadowTheAge** [comment]
Ah, yes, I misunderstool you. But maybe you clicked there to link that item or check belts amount whatever I will add to that tooltip later. Click that usually does one thing but in some case that you don't know in advance does something completely different isn't great user experience.

Next build will also feature "favourite" recipe. You add recipe to "favourites" manually. It will override any heuristics there is. In fact that is is implemented quite a long ago, I just can't get together to make a release.

**Comment by LoneWolfProductions** [comment]
>Click that usually does one thing but in some case that you don't know in advance does something completely different isn't great user experience.> 

I agree thats why i suggested to make it an option in the settings that you can turn on/off (off by default ;) )
or any other way you think is better

**Comment by ShadowTheAge** [comment]
I will add:
ctrl+click: Add the recipe in case there is only one option or only one option added to favourites, excluding barreling and voiding, otherwise show tooltip.

I may change the hotkey in case I will find better use for ctrl+click
